app/src/main/java/com/example/weather/MainActivity.java

```diff
@@ -2,13 +2,173 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.os.AsyncTask;
 import android.os.Bundle;
+import android.util.JsonReader;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import javax.net.ssl.HttpsURLConnection;
 
 public class MainActivity extends AppCompatActivity {
 
+    private TextView msgTemperature;
+    private EditText msgCity;
+    private Button buttonByCity;
+    private Button buttonByLocation;
+
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        msgTemperature = findViewById(R.id.msg_temperature);
+        msgCity = findViewById(R.id.msg_city);
+        buttonByCity = findViewById(R.id.button_by_city);
+        buttonByLocation = findViewById(R.id.button_by_location);
+    }
+
+    public void onClickByCity(View view) {
+        Log.d("myTag", "OnClick start");
+        String city = msgCity.getText().toString();
+        if (!checkCityString(city)) {
+            msgOnWrongCity();
+            return;
+        }
+
+        String request = formatRequestWithCity(city);
+        Log.d("myTag", request);
+        String response = doRequest(request);
+        if (response != null) {
+            Log.d("myTag", "len:" + response.length());
+        } else {
+            msgOnWrongRequest();
+            return;
+        }
+
+        String formattedOutput = getTemperature(response);
+        if (formattedOutput == null) {
+            msgOnWrongCity();
+            return;
+        }
+        String newMessage = getText(R.string.default_message) + " " + formattedOutput + " \u00B0C";
+        msgTemperature.setText(newMessage);
+        Log.d("myTag", "OnClick finish");
+    }
+
+    public void onClickByLocation(View view) {
+
+    }
+
+    private boolean checkCityString(String city) {
+        if (city == null) {
+            msgOnWrongCity();
+            return false;
+        }
+        else if (city.length() < 2 || city.length() > 30) {
+            msgOnWrongCity();
+            return false;
+        }
+        // TODO: do some other checking of input string
+        return true;
+    }
+
+    private String formatRequestWithCity(String city) {
+        String api_url = getText(R.string.weather_entry_point).toString();
```

Comment:
camelCase used the most in Java.
```suggestion
        String apiUrl = getText(R.string.weather_entry_point).toString();
```